From dbbf0956a5e7470771b572c40b0d3d97dc2f9b7d Mon Sep 17 00:00:00 2001
From: Cal Leeming <foxx@Cals-MacBook-Pro.local>
Date: Mon, 22 Aug 2016 15:53:36 +0100
Subject: [PATCH] Updated docs to use correct ssh-keygen command

Some versions of `ssh-keygen` will not output MD5 by default, including
the one which comes as default installed on MacOS El Capitan, resulting
in errors such as "error: Fingerprint format is not supported, or is
invalid". I've also removed the `tr` command because it causes a
superfluous '%' sign to be appended to the output on MacOS. This command
will unlikely be used in another shell script, and therefore shouldn't
be an issue.
---
 README.md | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/README.md b/README.md
index 2359168..63f2070 100644
--- a/README.md
+++ b/README.md
@@ -39,7 +39,11 @@ the environment variable equivalents:
   requests. If you use an SSH agent, the fingerprint is shown in `ssh-add -l`
   output. You can calculate the fingerprint like this:
 
-        ssh-keygen -l -f ~/.ssh/id_rsa.pub | awk '{print $2}' | tr -d '\n'
+    ssh-keygen -l -f ~/.ssh/id_rsa.pub | awk '{print $2}'
+
+  On new versions of `ssh-keygen`, you have to specify which output format to use;
+
+    ssh-keygen -E md5 -lf  ~/.ssh/id_rsa.pub  | awk '{print $2}'
 
   Your matching SSH *private* key must be beside the ".pub" public key file
   in your "~/.ssh" dir.
-- 
2.21.0

